addons/project/tests/test_project_recurrence.py

```diff
@@ -569,15 +570,36 @@ def get_task_and_subtask_counts(domain):
             'project_id': self.project_recurring.id
         })
         domain = [('project_id', '=', self.project_recurring.id)]
-        with Form(parent_task.with_context({'tracking_disable': True})) as task_form:
-            with task_form.child_ids.new() as subtask_form:
-                subtask_form.name = 'Child task 1'
-        with Form(parent_task.with_context({'tracking_disable': True})) as task_form:
-            with task_form.child_ids.new() as subtask_form:
-                subtask_form.name = 'Child task 2 that have recurrence'
+        self.env['project.task'].create({
+            'name': 'Child task 1',
+            'parent_id': parent_task.id,
+        })
+        child_task_2 = self.env['project.task'].create({
+            'name': 'Child task 2 that have recurrence',
+            'parent_id': parent_task.id,
+        })
+        grand_child_task_1 = self.env['project.task'].create({
+            'name': 'Grandchild task 1 (recurrent)',
+            'parent_id': child_task_2.id,
+        })
+        grand_child_task_2 = self.env['project.task'].create({
+            'name': 'Grandchild task 2',
+            'parent_id': child_task_2.id,
+        })
+        grand_child_task_3 = self.env['project.task'].create({
+            'name': 'Grandchild task 3',
+            'parent_id': grand_child_task_2.id,
+        })
+        grand_child_task_4 = self.env['project.task'].create({
+            'name': 'Grandchild task 4',
+            'parent_id': grand_child_task_3.id,
+        })
+        self.env['project.task'].create({
+            'name': 'Grandchild task 5',
+            'parent_id': grand_child_task_4.id,
+        })
```

Comment:
Could you rearrange the lines so that they are located close to where the related negative diff is. Thanks 🙂